OpenRA.Mods.Common/Util.cs

```diff
@@ -175,5 +175,15 @@ public static int RandomDelay(World world, int[] range)
 
 			return world.SharedRandom.Next(range[0], range[1]);
 		}
+
+		public static WDist MinimumRequiredScanRadius(Ruleset rules, ActorInfo ai, bool onlyBlockers = false)
+		{
+			var validActors = onlyBlockers ? rules.Actors.Where(a => a.Value.TraitInfos<BlocksProjectilesInfo>().Any()).ToList()
+				: rules.Actors.Where(a => a.Value.TraitInfos<HealthInfo>().Any()).ToList();
+
+			var largestOuterHealthRadius = validActors.SelectMany(a => a.Value.TraitInfos<HealthInfo>()).Max(h => h.Shape.OuterRadius);
+
+			return largestOuterHealthRadius;
```

Comment:
Memoization could be used here to avoid recalculating at least once for every actor type:
```c#
		static WDist blockerMinimumRequiredScanRadius;
		static WDist allMinimumRequiredScanRadius;
		static Ruleset lastRules = null;

		public static WDist MinimumRequiredScanRadius(Ruleset rules, ActorInfo ai, bool onlyBlockers = false)
		{
			if (lastRules != rules)
			{
				blockerMinimumRequiredScanRadius = rules.Actors.Where(a => a.Value.HasTraitInfo<BlocksProjectilesInfo>()).SelectMany(a => a.Value.TraitInfos<HealthInfo>()).Max(h => h.Shape.OuterRadius);
				allMinimumRequiredScanRadius = rules.Actors.SelectMany(a => a.Value.TraitInfos<HealthInfo>()).Max(h => h.Shape.OuterRadius);
				lastRules = rules;
			}

			return onlyBlockers ? blockerMinimumRequiredScanRadius : allMinimumRequiredScanRadius;
		}
```